CONTRIBUTING.md

```diff
@@ -0,0 +1,274 @@
+# Contributing
+
+As a contributor, here are the guidelines we would like you to follow:
+
+- [Question or Problem?](#question)
+- [Issues and Bugs](#issue)
+- [Submission Guidelines](#submit)
+- [Coding Rules](#rules)
+- [Commit Message Guidelines](#commit)
+
+
+## <a name="question"></a> Got a Question or Problem?
+
+Ask in `#dev-chat` in slack or post in `#its-not-working`.
+
+
+## <a name="issue"></a> Found a Bug?
+
+File an issue in JIRA under the appropriate project and send a message in `#dev-chat` on slack.
+
+Optionally you could just send a message in slack and an issue will hopefully be created for you.
+
+
+## <a name="submit"></a> Submission Guidelines
+
+
+### <a name="submit-issue"></a> Submitting an Issue
+
+Before you submit an issue, please search the `#its-not-working` channel as well as JIRA issues, maybe an issue for your
+problem already exists and the discussion might inform you of workarounds readily available.
+
+We want to fix all the issues as soon as possible, but before fixing a bug we need to reproduce and confirm it.
+In order to reproduce bugs, we require that you provide a minimal reproduction.
+Having a minimal reproducible scenario gives us a wealth of important information without going back and forth to you
+with additional questions.
+
+A minimal reproduction allows us to quickly confirm a bug (or point out a coding problem) as well as confirm that we
+are fixing the right problem.
+
+We require a minimal reproduction to save maintainers' time and ultimately be able to fix more bugs.
+Often, developers find coding problems themselves while preparing a minimal reproduction.
+
+Unfortunately, we are not able to investigate / fix bugs without a minimal reproduction, so if we don't hear back from
+you, we are going to close an issue that doesn't have enough info to be reproduced.
+
+### <a name="submit-pr"></a> Submitting a Pull Request (PR)
+
+Before you submit your Pull Request (PR) consider the following guidelines:
+
+1. Search github for an open or closed PR that relates to your submission.
+   You don't want to duplicate existing efforts.
+
+2. Be sure that an issue describes the problem you're fixing, or documents the design for the feature you'd like to add.
+   Discussing the design upfront helps to ensure that we're ready to accept your work.
+
+3. Create your branch `plat-1234-short-description` where `plat-1234` is your jira issue reference.
+
+4. Create your patch, **including appropriate test cases**.
+
+5. Follow our [Coding Rules](#rules).
+
+6. Commit your changes using a descriptive commit message that follows our [commit message conventions](#commit).
+   Adherence to these conventions is necessary because release notes are automatically generated from these messages.
+
+     ```shell
+     git commit --all
+     ```
+   Note: the optional commit `-a` command line option will automatically "add" and "rm" edited files.
```

Comment:
**[BREAKING]** I feel this is a bit confusing since it shows --all but then talks about -a

Reply:
Fixed